apps/api/src/app/environments-v1/usecases/output-renderers/expand-email-editor-schema.usecase.ts

```diff
@@ -92,60 +135,74 @@ export class ExpandEmailEditorSchemaUsecase {
+   *   ]
+   * }
+   *
+   * variables:
+   * { payload: { comments: [{ author: 'John Doe' }, { author: 'Jane Doe' }] } }
+   *
+   * result:
+   * [
+   *   { type: 'text', text: '{{ payload.comments[0].author }}' },
+   *   { type: 'text', text: '{{ payload.comments[1].author }}' }
+   * ]
+   *
+   */
+  private multiplyForEachNode(
+    node: TipTapNode & { attrs: { each: unknown } },
+    variables: Record<string, unknown>
+  ): TipTapNode[] {
+    const iterablePath = node.attrs.each as string;
+    const nodeContent = node.content || [];
+    const expandedContent: TipTapNode[] = [];
 
-    return this.regularExpansion(eachObject, templateContent);
-  }
+    const iterableArray = this.getValueByPath(variables, iterablePath) as [{ [key: string]: unknown }];
+
+    for (const [index] of iterableArray.entries()) {
+      const contentToExpand =
+        (this.isOrderedList(nodeContent) || this.isBulletList(nodeContent)) && nodeContent[0].content
+          ? nodeContent[0].content
+          : nodeContent;
 
-  private removeNodeFromParent(node: TipTapNode, parentNode?: TipTapNode) {
-    if (parentNode && parentNode.content) {
-      parentNode.content.splice(parentNode.content.indexOf(node), 1);
+      const hydratedContent = this.addIndexesToPlaceholders(contentToExpand, iterablePath, index);
+      expandedContent.push(...hydratedContent);
     }
-  }
 
-  private async hideShowIfNeeded(
-    variables: Record<string, unknown>,
-    node: TipTapNode & { attrs: { [MailyAttrsEnum.SHOW_IF_KEY]: unknown } },
-    parentNode?: TipTapNode
-  ): Promise<void> {
-    const { [MailyAttrsEnum.SHOW_IF_KEY]: showIfKey } = node.attrs;
+    return expandedContent;
+  }
 
-    if (showIfKey === undefined) {
-      return;
-    }
+  private addIndexesToPlaceholders(nodes: TipTapNode[], iterablePath: string, index: number): TipTapNode[] {
+    return nodes.map((node) => {
+      const newNode: TipTapNode = { ...node };
 
-    const parsedShowIfValue = await parseLiquid(showIfKey as string, variables);
-    const showIfValueBoolean =
-      typeof parsedShowIfValue === 'boolean' ? parsedShowIfValue : this.stringToBoolean(parsedShowIfValue);
+      if (this.isAVariableNode(newNode)) {
+        const nodePlaceholder = newNode.text as string;
+
+        /**
+         * example:
+         * iterablePath = payload.comments
+         * nodePlaceholder = {{ payload.comments.author }}
+         * text = {{ payload.comments[0].author }}
+         */
+        newNode.text = nodePlaceholder.replace(iterablePath, `${iterablePath}[${index}]`);
+        newNode.type = 'text'; // set 'variable' to 'text' to for Liquid to recognize it
+      } else if (newNode.content) {
+        newNode.content = this.addIndexesToPlaceholders(newNode.content, iterablePath, index);
+      }
 
-    if (!showIfValueBoolean) {
-      this.removeNodeFromParent(node, parentNode);
-    } else {
-      delete node.attrs[MailyAttrsEnum.SHOW_IF_KEY];
-    }
+      return newNode;
+    });
```

Comment:
Instead of replacing for loop values directly in this usecase as it was before, we create just placeholders  `{{ payload.comments[0].author }}` and leave the actual value substitution for Liquid parser down the line